modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -843,6 +971,297 @@ static void run_arithm_s(DST out[], const SRC in[], int width, int chan,
+{
+    constexpr int nlanes = static_cast<int>(v_uint8::nlanes);
+
+    if (length < nlanes)
+        return 0;
+
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            v_float32 a1 = v_setr(in + x),
+                      a2 = v_setr(in + x + nlanes / 4),
+                      a3 = v_setr(in + x + nlanes / 2),
+                      a4 = v_setr(in + x + 3 * nlanes / 4);
+
+            vx_store(&out[x], v_pack_u(v_pack(v_round(v_absdiff(a1, s)),
+                                              v_round(v_absdiff(a2, s))),
+                                       v_pack(v_round(v_absdiff(a3, s)),
+                                              v_round(v_absdiff(a4, s)))));
+        }
+
+        if (x < length && (in != out))
+        {
+            x = length - nlanes;
+            continue;  // process one more time (unaligned tail)
+        }
+        break;
+    }
+    return x;
+}
+
+CV_ALWAYS_INLINE void absdiffc_short_store_c1c2c4(short* out_ptr, const v_int32& c1, const v_int32& c2)
+{
+    vx_store(out_ptr, v_pack(c1, c2));
+}
+
+CV_ALWAYS_INLINE void absdiffc_short_store_c1c2c4(ushort* out_ptr, const v_int32& c1, const v_int32& c2)
+{
+    vx_store(out_ptr, v_pack_u(c1, c2));
+}
+
+template<typename T>
+CV_ALWAYS_INLINE int absdiffc_simd_shorts_c1c2c4(const T in[], T out[],
+                                                 const v_float32& s, const int length)
+{
+    constexpr int nlanes = static_cast<int>(v_uint16::nlanes);
+    if (length < nlanes)
+        return 0;
+
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            v_float32 a1 = v_setr(in + x),
+                      a2 = v_setr(in + x + nlanes/2);
+
+            absdiffc_short_store_c1c2c4(&out[x], v_round(v_absdiff(a1, s)),
+                                                 v_round(v_absdiff(a2, s)));
+        }
+
+        if (x < length && (in != out))
+        {
+            x = length - nlanes;
+            continue;  // process one more time (unaligned tail)
+        }
+        break;
+    }
+    return x;
+}
+
+template<typename T>
+CV_ALWAYS_INLINE int absdiffc_simd_cases(const T in[], const v_float32& s, T out[], const int length)
+{
+    if (std::is_same<T, uchar>::value)
+    {
+        return absdiffc_simd_uchar_c1c2c4(reinterpret_cast<const uchar*>(in),
+                                          reinterpret_cast<uchar*>(out), s, length);
+    }
+    else if (std::is_same<T, ushort>::value)
+    {
+        return absdiffc_simd_shorts_c1c2c4(reinterpret_cast<const ushort*>(in),
+                                           reinterpret_cast<ushort*>(out), s, length);
+    }
+    else if (std::is_same<T, short>::value)
+    {
+        return absdiffc_simd_shorts_c1c2c4(reinterpret_cast<const short*>(in),
+                                           reinterpret_cast<short*>(out), s, length);
+    }
```

Comment:
You can name `absdiffc_simd_uchar_c1c2c4` and `absdiffc_simd_shorts_c1c2c4` with a single name and utilize the function overload here, actually this `absdiffc_simd_cases` function will become redundant since this dispatching will be done automatically (uchar case should be done as a template specialization)

Reply:
Done.